ras_party/controllers/account_controller.py

```diff
@@ -417,22 +417,46 @@ def compensate_oauth_change():
 
 
 @with_db_session
-def resend_verification_email(party_uuid, session):
+def resend_verification_email_by_uuid(party_uuid, session):
     """
-    Check and resend an email verification email
+    Check and resend an email verification email using the party id
     :param party_uuid: the party uuid
-    :return: make_response
+    :param session: database session
+    :return: response
     """
     logger.debug('Attempting to resend verification_email', party_uuid=party_uuid)
 
     respondent = query_respondent_by_party_uuid(party_uuid, session)
     if not respondent:
         raise ClientError(NO_RESPONDENT_FOR_PARTY_ID, status=404)
 
+    response = _resend_verification_email(respondent)
+    return response
+
+
+@with_db_session
+def resend_verification_email_expired_token(token, session):
+    """
+    Check and resend an email verification email using the expired token
+    :param token: the expired token
+    :param session: database session
+    :return: response
+    """
+    email_address = decode_email_token(token, duration=None)
+    respondent = query_respondent_by_email(email_address, session)
+
+    if not respondent:
+        raise ClientError("Respondent does not exist", status=404)
+
+    response = _resend_verification_email(respondent)
+    return response
+
+
+def _resend_verification_email(respondent):
```

Comment:
Nice use of common functionality, we should do this more often in our services 👍